controllers/ibmblockcsi_controller_test.go

```diff
@@ -0,0 +1,242 @@
+				  err := k8sClient.Get(context.Background(), saKey, sa)
+				  return sa, err
+				}, timeout, interval).ShouldNot(BeNil())
+
+				By("Getting ClusterRole")
+				cr := &rbacv1.ClusterRole{}
+				clusterRoles := []config.ResourceName{config.ExternalProvisionerClusterRole, config.ExternalAttacherClusterRole, 
+				config.ExternalSnapshotterClusterRole, config.ExternalResizerClusterRole, config.CSIAddonsReplicatorClusterRole,
+				config.CSIControllerSCCClusterRole, config.CSINodeSCCClusterRole}
+
+				for _, clusterRole := range clusterRoles {
+					crKey := types.NamespacedName{
+						Name:      config.GetNameForResource(clusterRole, found.Name),
+						Namespace: "",
+					}
+					Eventually(func() (*rbacv1.ClusterRole, error) {
+						err := k8sClient.Get(context.Background(), crKey, cr)
+						return cr, err
+					}, timeout, interval).ShouldNot(BeNil())
+				}
+
+				By("Getting ClusterRoleBinding")
+				crb := &rbacv1.ClusterRoleBinding{}
+				clusterRoleBindings := []config.ResourceName{config.ExternalProvisionerClusterRoleBinding,
+					config.ExternalAttacherClusterRoleBinding, config.ExternalSnapshotterClusterRoleBinding,
+					config.ExternalResizerClusterRoleBinding, config.CSIAddonsReplicatorClusterRoleBinding,
+					config.CSIControllerSCCClusterRoleBinding, config.CSINodeSCCClusterRoleBinding}
+
+				for _, clusterRoleBinding := range clusterRoleBindings {
+					crbKey := types.NamespacedName{
+						Name:      config.GetNameForResource(clusterRoleBinding, found.Name),
+						Namespace: "",
+					}
+					Eventually(func() (*rbacv1.ClusterRoleBinding, error) {
+						err := k8sClient.Get(context.Background(), crbKey, crb)
+						return crb, err
+					  }, timeout, interval).ShouldNot(BeNil())
+				}
+
+				By("Getting controller StatefulSet")
+				controller := &appsv1.StatefulSet{}
+				controllerKey := types.NamespacedName{
+				  Name:      config.GetNameForResource(config.CSIController, found.Name),
+				  Namespace: found.Namespace,
+				}
+				Eventually(func() (*appsv1.StatefulSet, error) {
+				  err := k8sClient.Get(context.Background(), controllerKey, controller)
+				  return controller, err
+				}, timeout, interval).ShouldNot(BeNil())
+				Expect(len(controller.Spec.Template.Spec.Containers)).To(BeNumerically(">", 0))
+				for _, container := range controller.Spec.Template.Spec.Containers {
+					image, ok := containersImages[container.Name]
+					Expect(ok).To(BeTrue(), fmt.Sprintf("container %s not found in %s", container.Name, containersImages))
+					Expect(container.Image).To(Equal(image))
+				}
+
+
+				By("Getting node DaemonSet")
+				node := &appsv1.DaemonSet{}
+				nodeKey := types.NamespacedName{
+				  Name:      config.GetNameForResource(config.CSINode, found.Name),
+				  Namespace: found.Namespace,
+				}
+				Eventually(func() (*appsv1.DaemonSet, error) {
+				  err := k8sClient.Get(context.Background(), nodeKey, node)
+				  return node, err
+				}, timeout, interval).ShouldNot(BeNil())
```

Comment:
please try to go over the PR diff and see if you spot any code dups, then move them to common locations

Reply:
done as much as I can